src/library/scala/util/matching/Regex.scala

```diff
@@ -219,15 +219,18 @@ class Regex private[matching](val pattern: Pattern, groupNames: String*) extends
    *  val namedYears = for (m <- namedDate findAllMatchIn dates) yield m group "year"
    *  }}}
    *
-   *  Group names supplied to the constructor are preferred to inline group names
-   *  when retrieving matched groups by name. Not all platforms support inline names.
+   *  Inline group names are preferred over group names supplied to the constructor
+   *  when retrieving matched groups by name. Group names supplied to the constructor
+   *  should be considered deprecated.
    *
    *  This constructor does not support options as flags, which must be
-   *  supplied as inline flags in the pattern string: `(?idmsux-idmsux)`.
+   *  supplied as inline flags in the pattern string: `(?idmsuxU)`.
```

Comment:
I haven't refreshed my regex syntax yet, but iirc the `-` was a reminder that you can turn off flags. I'll have to read the doc for `U`. The comment suggests that the string representation is canonical; that is a good thing.

Reply:
The U is for `UNICODE_CHARACTER_CLASSES` (u is for `UNICODE_CASE`).

Sure you can turn off flags, but since you start with 0 flags enabled, it is always useless in this context.

Reply:
The code comment is that the couple of global flags which don't have inline equivalent can't be specified. (LITERAL, CANON_EQ.) Someone complained when the primary constructor taking a compiled Pattern was made private; this would be the use case. Inline flags can be turned on/off `"(?i)abc(?-i)abc".r` but the context here is what is enabled at construction. Probably the Scaladoc was written with Java 1.6 doc.